component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -935,7 +935,14 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
         String encodedMobileNo = encoder.encode(mobile);
         smsUrl = smsUrl.replaceAll("\\$ctx.num", encodedMobileNo).replaceAll("\\$ctx.msg",
                 smsMessage.replaceAll("\\s", "+") + otpToken);
-        URL smsProviderUrl = new URL(smsUrl);
+        URL smsProviderUrl = null;
+        try {
+            smsProviderUrl = new URL(smsUrl);
+        } catch (MalformedURLException e) {
+            log.error("Invalid SMS URL provided : " + e);
```

Comment:
`log.error("Error while parsing SMS provider URL: " + smsUrl, e);`

Reply:
Changed the error message as avove